### PR TITLE
[12.0][NEW] módulo para atualizar cotações de moedas

### PR DESCRIPTION
**Pull request by renatonlima**
Este PR cria um novo módulo chamado l10n_br_currency_update_rate que localiza o módulo currency_update_rate do repositório [OCA/currency](http://github.com/OCA/currency)

Este módulo consulta o webservice fornecido pelo [Banco Central do Brasil](https://dadosabertos.bcb.gov.br/dataset/taxas-de-cambio-todos-os-boletins-diarios) para fazer a atualização das cotações de moedas automaticamente.

### CODE REVIEW

File: l10n_br_currency_rate_update/__init__.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/__init__.py:2]: Missing license

File: l10n_br_currency_rate_update/__manifest__.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/__manifest__.py]: Base version is `12.0.1.0.0`

File: l10n_br_currency_rate_update/__manifest__.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/__manifest__.py]: Empty data is not needed

File: l10n_br_currency_rate_update/models/__init__.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/models/__init__.py:2]: License missing

File: l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py:20]: This may be "optimized" as in https://github.com/OCA/currency/blob/12.0/currency_rate_update/models/res_currency_rate_provider_ECB.py to avoid extra padding

File: l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py:34]: This may be "optimized" as in https://github.com/OCA/currency/blob/12.0/currency_rate_update/models/res_currency_rate_provider_ECB.py to avoid extra padding

File: l10n_br_currency_rate_update/tests/__init__.py
Comment (alexey-pelykh) [inline l10n_br_currency_rate_update/tests/__init__.py:2]: License missing

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
Minor formatting issues, otherwise code LGTM

File: l10n_br_currency_rate_update/tests/__init__.py
Comment (rvalyi) [reply l10n_br_currency_rate_update/tests/__init__.py:2]: @alexey-pelykh 
Thank you very much for the review. We however don't put license headers anymore in these `__init__.py` almost empty files at the OCA to avoid bloating the code.

Comment (alexey-pelykh) [reply l10n_br_currency_rate_update/tests/__init__.py:2]: There's a template https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py
Comment (renatonlima) [reply l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py:20]: Done!

File: l10n_br_currency_rate_update/tests/__init__.py
Comment (rvalyi) [reply l10n_br_currency_rate_update/tests/__init__.py:2]: oh you are right indeed. I mean we don't put the copyrights in `__init__.py `anymore but indeed the single line license header is usually set.

File: l10n_br_currency_rate_update/models/__init__.py
Comment (renatonlima) [reply l10n_br_currency_rate_update/models/__init__.py:2]: Done!

File: l10n_br_currency_rate_update/tests/__init__.py
Comment (renatonlima) [reply l10n_br_currency_rate_update/tests/__init__.py:2]: Done!

File: l10n_br_currency_rate_update/__manifest__.py
Comment (renatonlima) [reply l10n_br_currency_rate_update/__manifest__.py]: Done!

File: l10n_br_currency_rate_update/__manifest__.py
Comment (renatonlima) [reply l10n_br_currency_rate_update/__manifest__.py]: Done!

File: l10n_br_currency_rate_update/__init__.py
Comment (renatonlima) [reply l10n_br_currency_rate_update/__init__.py:2]: Done!

**Comment by renatonlima** [comment]
@alexey-pelykh thank you so much for your review!

**Comment by renatonlima** [comment]
cc @mileo @hendrixcosta @mbcosta

File: l10n_br_currency_rate_update/__init__.py
Comment (alexey-pelykh) [reply l10n_br_currency_rate_update/__init__.py:2]: @renatonlima missed here

File: l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py
Comment (alexey-pelykh) [reply l10n_br_currency_rate_update/models/res_currency_rate_provider_bcb.py:20]: Not sure I follow, I meant:
```
self.ensure_one()
if self.service != 'BCB':
    return super()._get_supported_currencies()

here_goes_your_code
```

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: oca_dependencies.txt
Comment (hendrixcosta) [inline oca_dependencies.txt:1]: Talvez não precisaria desse arquivo.

**Review by hendrixcosta — APPROVED** [review]

File: oca_dependencies.txt
Comment (renatonlima) [reply oca_dependencies.txt:1]: Este arquivo é para explicitar a dependência do módulo currency_update_rate que esta no repo [OCA/currency](http://github.com/OCA/currency)

Comment (hendrixcosta) [reply oca_dependencies.txt:1]: Justo. Não me atentei a última linha, só vi a parte comentada.

Comment (alexey-pelykh) [reply oca_dependencies.txt:1]: That commented part isn't really needed, if take a look on other repositories

Comment (renatonlima) [reply oca_dependencies.txt:1]: I removed commented part!

**Review by alexey-pelykh — APPROVED** [review]
Code LGTM, yet there can be visual enhancement over here https://github.com/OCA/l10n-brazil/pull/683/files#r282373946

**Review by hendrixcosta — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
Before merging - please squash the commits into one

**Review by mileo — APPROVED** [review]

**Pull request merged**
